TestSolutions/Lombiq.Analyzers.PackageReference/IndirectPackageReference/IndirectPackageReference.csproj

```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+  </PropertyGroup>
+
+  <!-- This project demonstrates that the package reference doesn't transfer the analyzer effect to its dependents. If
+       you set ImportPackageAgain to "true" and run static code analysis on this soltuion you will get twice as many
+       warnings. This is the desired behavior, shows that the analyzers only activate when they are explicitly imported.
+  -->
+
+  <ItemGroup Condition="'$(ImportPackageAgain)' == 'true'">
+    <PackageReference Include="Lombiq.Analyzers" Version="2.1.1-osoe-110.14" />
+  </ItemGroup>
+  
+  <!-- Dummy target for when ImportPackageAgain is disabled. -->
+  <Target Name="LombiqNetAnalyzers" Condition="'$(ImportPackageAgain)' != 'true'">
+  </Target>
+
+  <Target Name="CopyAnalyzerViolations" DependsOnTargets="LombiqNetAnalyzers" AfterTargets="LombiqNetAnalyzers">
+    <Copy SourceFiles="..\..\AnalyzerViolations.cs" DestinationFolder="$(MSBuildThisFileDirectory)" />
```

Comment:
When I open the `Lombiq.Analyzers.PackageReference` solution in VS, the `AnalyzerViolations.cs` file is not copied into this project. It looks like I have to call the `LombiqNetAnalyzers` target to make that happen.

Is all of this only there for continuous integration tests? Where does this actually come into play? Is it [this line](https://github.com/Lombiq/GitHub-Actions/pull/21/files#diff-4cf358b6f80e00ce9cd739a04c94dfad647b09ca5d6a524757944a1c611d6848R54) that triggers the copying and then the solution build won't show any errors on the _AnalyzerViolations.cs_ file inside this project? But if there's no (indirect) reference to `Analyzers`, then that seems pointless. What am I missing?

Reply:
I can't reproduce that. When I open the solution in VS it immediately create copies the files before build so you don't have to do anything special. Similar for Rider too. I guess this is due to incremental compilation during build or something along those lines.

https://user-images.githubusercontent.com/4007293/182661120-131c33f3-c7f1-47b0-9a4a-bc0d8ba4e51a.mp4

The separate `LombiqNetAnalyzers` target is only for CI, so you don't have to build twice.

Reply:
I missed to include the `AfterTargets="AfterResolveReferences"` here as well to splice it into the normal build process like it is in Lombiq.Analyzers/Build.props.